Exception when exporting with configured credentials

Using 0.9a version, once even a single credentials username/password credentials has been configured, it makes CasC exports fail with this exception.

```
Jul 05, 2018 8:11:34 AM org.jenkinsci.remoting.util.AnonymousClassWarnings warn
WARNING: Attempt to (de-)serialize anonymous class com.cloudbees.plugins.credentials.CredentialsScope$2 in file:/var/jenkins_home/plugins/credentials/WEB-INF/lib/credentials.jar; see: https://jenkins.io/redirect/serialization-of-anonymous-classes/
Jul 05, 2018 8:11:49 AM org.eclipse.jetty.server.handler.ContextHandler$Context log
WARNING: Error while serving http://localhost:8080/configuration-as-code/export
java.lang.reflect.InvocationTargetException
        at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:347)
        at org.kohsuke.stapler.interceptor.RequirePOST$Processor.invoke(RequirePOST.java:77)
        at org.kohsuke.stapler.PreInvokeInterceptedFunction.invoke(PreInvokeInterceptedFunction.java:26)
        at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:184)
        at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:117)
        at org.kohsuke.stapler.MetaClass$1.doDispatch(MetaClass.java:129)
        at org.kohsuke.stapler.NameBasedDispatcher.dispatch(NameBasedDispatcher.java:58)
        at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:715)
        at org.kohsuke.stapler.Stapler.invoke(Stapler.java:845)
        at org.kohsuke.stapler.MetaClass$10.dispatch(MetaClass.java:374)
        at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:715)
        at org.kohsuke.stapler.Stapler.invoke(Stapler.java:845)
        at org.kohsuke.stapler.Stapler.invoke(Stapler.java:649)
        at org.kohsuke.stapler.Stapler.service(Stapler.java:238)
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:860)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
        at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:154)
        at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:157)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
        at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:64)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
        at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
        at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:90)
        at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:171)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
        at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:49)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
        at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:82)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
        at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
        at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1637)
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:533)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:524)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:190)
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1595)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:188)
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1253)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:168)
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:473)
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1564)
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:166)
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1155)
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:132)
        at org.eclipse.jetty.server.Server.handle(Server.java:530)
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:347)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:256)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:279)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:102)
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:124)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:247)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.produce(EatWhatYouKill.java:140)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:131)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:382)
        at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: argument type mismatch
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at org.jenkinsci.plugins.casc.DataBoundConfigurator.describe(DataBoundConfigurator.java:218)
        at org.jenkinsci.plugins.casc.HeteroDescribableConfigurator.describe(HeteroDescribableConfigurator.java:129)
        at org.jenkinsci.plugins.casc.HeteroDescribableConfigurator.describe(HeteroDescribableConfigurator.java:35)
        at org.jenkinsci.plugins.casc.Attribute.describe(Attribute.java:175)
        at org.jenkinsci.plugins.casc.DataBoundConfigurator.describe(DataBoundConfigurator.java:227)
        at org.jenkinsci.plugins.casc.Attribute.describe(Attribute.java:175)
        at org.jenkinsci.plugins.casc.credentials.SystemCredentialsProviderConfigurator.describe(SystemCredentialsProviderConfigurator.java:51)
        at org.jenkinsci.plugins.casc.credentials.SystemCredentialsProviderConfigurator.describe(SystemCredentialsProviderConfigurator.java:21)
        at org.jenkinsci.plugins.casc.Attribute.describe(Attribute.java:179)
        at org.jenkinsci.plugins.casc.credentials.CredentialsRootConfigurator.describe(CredentialsRootConfigurator.java:76)
        at org.jenkinsci.plugins.casc.credentials.CredentialsRootConfigurator.describe(CredentialsRootConfigurator.java:37)
        at org.jenkinsci.plugins.casc.ConfigurationAsCode.doExport(ConfigurationAsCode.java:234)
        at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
        at org.kohsuke.stapler.Function$MethodFunction.invoke(Function.java:343)
        ... 60 more
```

--- a/plugin/src/main/java/org/jenkinsci/plugins/casc/DataBoundConfigurator.java
+++ b/plugin/src/main/java/org/jenkinsci/plugins/casc/DataBoundConfigurator.java
@@ -6,6 +6,7 @@ import org.apache.commons.lang.StringUtils;
 import org.jenkinsci.plugins.casc.model.CNode;
 import org.jenkinsci.plugins.casc.model.Mapping;
 import org.kohsuke.stapler.ClassDescriptor;
+import org.kohsuke.stapler.Stapler;
 
 import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
@@ -255,7 +256,7 @@ public class DataBoundConfigurator<T> extends BaseConfigurator<T> {
         for (int i = 0; i < parameters.length; i++) {
             final Parameter p = parameters[i];
             final Attribute a = detectActualType(names[i], TypePair.of(p));
-            args[i] = a.getValue(instance);
+            args[i] = Stapler.CONVERT_UTILS.convert(a.getValue(instance), a.type);
             if (args[i] == null && p.getType().isPrimitive()) {
                 args[i] = defaultValue(p.getType());
             }

--- a/plugin/src/test/java/org/jenkinsci/plugins/casc/SystemCredentialsTest.java
+++ b/plugin/src/test/java/org/jenkinsci/plugins/casc/SystemCredentialsTest.java
@@ -9,6 +9,7 @@ import jenkins.model.Jenkins;
 import org.jenkinsci.plugins.casc.misc.ConfiguredWithCode;
 import org.jenkinsci.plugins.casc.misc.EnvVarsRule;
 import org.jenkinsci.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import org.jenkinsci.plugins.casc.model.CNode;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.RuleChain;
@@ -23,6 +24,7 @@ import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertThat;
 
 /**
@@ -52,7 +54,11 @@ public class SystemCredentialsTest {
                 UsernamePasswordCredentials.class, jenkins, ACL.SYSTEM, Collections.emptyList()
         );
         assertThat(ups, hasSize(1));
-        assertThat(ups.get(0).getPassword().getPlainText(), equalTo("1234"));
+        final UsernamePasswordCredentials up = ups.get(0);
+        assertThat(up.getPassword().getPlainText(), equalTo("1234"));
+        final CNode node = Configurator.lookup(up.getClass()).describe(up);
+        assertEquals("1234", node.asMapping().getScalarValue("password"));
+
 
         List<CertificateCredentials> certs = CredentialsProvider.lookupCredentials(
                 CertificateCredentials.class, jenkins, ACL.SYSTEM, Collections.emptyList()
@@ -71,5 +77,7 @@ public class SystemCredentialsTest {
             assertThat(logRecord.getMessage(), not(containsString("1234")));
             assertThat(logRecord.getMessage(), not(containsString("ABCD")));
         }
+
+
     }
 }

